contracts/token/ERC721/ERC721.sol

```diff
@@ -257,6 +259,46 @@ abstract contract ERC721 is Context, ERC165, IERC721, IERC721Metadata, IERC721Er
         }
     }
 
+    /**
+     * @dev Transfers `tokenId` from its current owner to `to`, or alternatively mints (or burns) if the current owner
+     * (or `to`) is the zero address.
+     *
+     * The `constraints` argument is used to specify constraints, and eventually revert. For example this can be used
+     * to ensure that the current owner is what was expected.
+
+     * All customizations to transfers, mints, and burns should be done by overriding this function.
+     *
+     * Emits a {Transfer} event.
+     */
+    function _update(
+        address to,
+        uint256 tokenId,
+        function(address, address, uint256) view constraints
+    ) internal virtual returns (address) {
+        address from = _ownerOf(tokenId);
+
+        constraints(from, to, tokenId);
+
+        if (from != address(0)) {
+            delete _tokenApprovals[tokenId];
+            unchecked {
+                _balances[from] -= 1;
+            }
+        }
+
+        if (to != address(0)) {
+            unchecked {
+                _balances[to] += 1;
```

Comment:
Resolving to focus the conversation on the other similar comment.

Reply:
I'd keep it open so we have the suggestion, and we don't forget about it. But yes, the discussion should happen in the other conversation.

Reply:
We agreed to keep it as it is because it's not a highly meaningful change. We might reconsider once this PR is done so we can correctly measure the gas savings.